cmd/ddblibrarian-clone/main.go

```diff
@@ -0,0 +1,160 @@
+package main
+
+import (
+	"flag"
+	"log"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/dynamodb"
+	"github.com/marcoalmeida/ddblibrarian"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"fmt"
+
+)
+
+type appConfig struct {
+	srcRegion        string
+	dstRegion        string
+	srcTable         string
+	dstTable         string
+	partitionKey     string
+	partitionKeyType string
+	rangeKey         string
+	rangeKeyType     string
+	snapshot         string
+}
+
+func checkFlags(app *appConfig) {
+	if app.srcTable == "" || app.dstTable == "" {
+		log.Fatal("Both source and destination tables are mandatory")
+	}
+
+	if app.partitionKey == "" {
+		log.Fatal("The partition key is required")
+	}
+
+	if app.partitionKeyType == "" {
+		log.Fatal("The partition key type (S or N) is required")
+	}
+}
+
+func connect(app *appConfig) (*dynamodb.DynamoDB, *ddblibrarian.Library) {
+	srcSession, err := session.NewSession(&aws.Config{
+		Region:     aws.String(app.srcRegion),
+		MaxRetries: aws.Int(3),
+	})
+	if err != nil {
+		log.Fatal(err.Error())
+	}
+
+	dstSession, err := session.NewSession(&aws.Config{
+		Region:     aws.String(app.dstRegion),
+		MaxRetries: aws.Int(3),
+	})
+	if err != nil {
+		log.Fatal(err.Error())
+	}
+
+	librarian, err := ddblibrarian.New(
+		app.dstTable,
+		app.partitionKey,
+		app.partitionKeyType,
+		app.rangeKey,
+		app.rangeKeyType,
+		dstSession,
+	)
+	if err != nil {
+		log.Fatal(err.Error())
+	}
+
+	return dynamodb.New(srcSession), librarian
+}
+
+func writeItems(table string,library *ddblibrarian.Library, items []map[string]*dynamodb.AttributeValue) {
+	requests := make(map[string][]*dynamodb.WriteRequest, 0)
+
+	for _, item := range items {
+		requests[table] = append(requests[table], &dynamodb.WriteRequest{
+			PutRequest: &dynamodb.PutRequest{
+				Item: item,
+			}})
+    }
+
+	_, err := library.BatchWriteItem(&dynamodb.BatchWriteItemInput{
+		RequestItems: requests,
+	})
+	if err != nil {
+		log.Println("Failed to write batch:", err)
```

Comment:
Add the list of keys that we failed to write as well? (considering that we are doing this async) So that we know exactly what keys haven't been copied over.

Reply:
Makes sense. I'll dump it to stderr.